src/components/Coins.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getCoins } from '../redux/coins/CoinsSlice';
+
+const Coins = () => {
+  const { coins, isLoading, errorMessage } = useSelector((state) => state.coins);
+  const dispatch = useDispatch();
+  const [searchText, setSearchText] = useState('');
+
+  useEffect(() => {
+    dispatch(getCoins());
+  }, [dispatch]);
+
+  const handleSearch = (e) => {
+    setSearchText(e.target.value);
+  };
+
+  const filteredCoins = coins.filter(
+    (coin) => coin.name.toLowerCase().includes(searchText.toLowerCase()),
+  );
+
+  return (
+    <section className="coin-section">
+      <div className="coin-topic">
+        <h2>Cryptocurrency Market</h2>
+        <p>
+          (
+          {filteredCoins.length}
+          {' '}
+          Famouse Crypto Coins)
+        </p>
+        <div className="search-bar-container">
+          <input type="text" placeholder="Search" className="search-bar" value={searchText} onChange={handleSearch} />
+        </div>
+      </div>
```

Comment:
- [ ] Please read the `design` guidelines comments and refactor your `design`, as per the guidelines your design should have nice `layout` composition and spacing but it seems like your design needs to have some space around the `element` to make it looks great and one more important thing is change the `font-family` for your `input` field placeholder too.

![image](https://github.com/tawakuliKH/coinCap-React/assets/66476329/e95e3c3c-f211-4f3c-9007-a503ad78ca77)